example/src/Client/utils/AccountsManager.ts

```diff
@@ -6,39 +6,50 @@ import {
   NoteEncrypted,
   Note,
 } from "@ironfish/rust-nodejs";
+import { BufferMap } from "buffer-map";
 import { BlockCache } from "./BlockCache";
 import {
   LightBlock,
+  LightSpend,
   LightTransaction,
 } from "../../../../src/models/lightstreamer";
 import { logThrottled } from "./logThrottled";
 
-export interface DecryptedNoteValue {
+export interface StoredNote {
   accountId: string;
   note: Note;
   spent: boolean;
   transactionHash: Buffer;
-  index: number | null;
-  nullifier: Buffer | null;
-  blockHash: Buffer | null;
-  sequence: number | null;
+  index: number;
+  nullifier: Buffer;
+  blockHash: Buffer;
+  sequence: number;
 }
 
+/* Note hash => StoredNote */
+type StoredNotesByNoteHash = BufferMap<StoredNote>;
+
+/* Asset ID => StoredNotesByNoteHash */
+type AssetContentByAssetId = BufferMap<StoredNotesByNoteHash>;
+
 interface AccountData {
   key: Key;
   head: number;
-  assets: Map<
-    string,
-    {
-      balance: bigint;
-      decryptedNotes: DecryptedNoteValue[];
-    }
-  >;
+  assets: AssetContentByAssetId;
```

Comment:
I think I would prefer long term for this example we store the asset inside the note rather than make it a map of asset -> notes. This way we could actually make the map block -> note, which would enable us deleting by blocks during reorg as well.

If we need a mapping, we could do a similar thing as below for the other maps.